src/link/script.c

```diff
@@ -0,0 +1,214 @@
+} bank[MAXBANKS];
+
+static int current_bank = -1; /* Bank as seen by the bank array */
+static int current_real_bank = -1; /* bank as seen by the GB */
+
+void script_InitSections(void)
+{
+	int i;
+	for (i = 0; i < MAXBANKS; i++) {
+		if (i == BANK_ROM0) {
+			/* ROM0 bank */
+			bank[i].address = 0x0000;
+			if (options & OPT_SMALL) {
+				bank[i].top_address = 0x8000;
+			} else {
+				bank[i].top_address = 0x4000;
+			}
+			bank[i].type = SECT_ROM0;
+		} else if (i >= BANK_ROMX && i < BANK_ROMX + BANK_COUNT_ROMX) {
+			/* Swappable ROM bank */
+			bank[i].address = 0x4000;
+			/*
+			 * Now, this shouldn't really be necessary... but for
+			 * good measure we'll do it anyway.
+			 */
+			if (options & OPT_SMALL) {
+				bank[i].top_address = 0x4000;
+			} else {
+				bank[i].top_address = 0x8000;
+			}
+			bank[i].type = SECT_ROMX;
+		} else if (i == BANK_WRAM0) {
+			/* WRAM */
+			bank[i].address = 0xC000;
+			if (options & OPT_CONTWRAM) {
+				bank[i].top_address = 0xE000;
+			} else {
+				bank[i].top_address = 0xD000;
+			}
+			bank[i].type = SECT_WRAM0;
+		} else if (i >= BANK_SRAM && i < BANK_SRAM + BANK_COUNT_SRAM) {
+			/* Swappable SRAM bank */
+			bank[i].address = 0xA000;
+			bank[i].top_address = 0xC000;
+			bank[i].type = SECT_SRAM;
+		} else if (i >= BANK_WRAMX && i < BANK_WRAMX + BANK_COUNT_WRAMX) {
+			/* Swappable WRAM bank */
+			bank[i].address = 0xD000;
+			bank[i].top_address = 0xE000;
+			bank[i].type = SECT_WRAMX;
+		} else if (i >= BANK_VRAM && i < BANK_VRAM + BANK_COUNT_VRAM) {
+			/* Swappable VRAM bank */
+			bank[i].address = 0x8000;
+			bank[i].top_address = 0xA000;
+			bank[i].type = SECT_VRAM;
+		} else if (i == BANK_OAM) {
+			/* OAM */
+			bank[i].address = 0xFE00;
+			bank[i].top_address = 0xFEA0;
+			bank[i].type = SECT_OAM;
+		} else if (i == BANK_HRAM) {
+			/* HRAM */
+			bank[i].address = 0xFF80;
+			bank[i].top_address = 0xFFFF;
+			bank[i].type = SECT_HRAM;
+		} else {
+			errx(1, "(INTERNAL) Unknown bank type!");
+		}
+	}
+}
+
+void script_SetCurrentSectionType(const char *type, unsigned int bank)
+{
+	if (strcmp(type, "ROM0") == 0) {
+		if (bank != 0)
+			errx(1, "(Internal) Trying to assign a bank number to ROM0.\n");
+		current_bank = BANK_ROM0;
+		current_real_bank = 0;
+		return;
+	} else if (strcmp(type, "ROMX") == 0) {
+		if (bank >= BANK_COUNT_ROMX)
+			errx(1, "ROMX index too big (%d >= %d).\n", bank, BANK_COUNT_ROMX);
+		current_bank = BANK_ROMX + bank - 1;
+		current_real_bank = bank;
+		return;
+	} else if (strcmp(type, "VRAM") == 0) {
+		if (bank >= BANK_COUNT_VRAM)
+			errx(1, "VRAM index too big (%d >= %d).\n", bank, BANK_COUNT_VRAM);
+		current_bank = BANK_VRAM + bank;
+		current_real_bank = bank;
+		return;
+	} else if (strcmp(type, "WRAM0") == 0) {
+		if (bank != 0)
+			errx(1, "(Internal) Trying to assign a bank number to WRAM0.\n");
+		current_bank = BANK_WRAM0;
+		current_real_bank = 0;
+		return;
+	} else if (strcmp(type, "WRAMX") == 0) {
+		if (bank >= BANK_COUNT_WRAMX)
```

Comment:
Right! Done. I didn't check see this because pokered doesn't use WRAMX up to bank 7...